lib/chef/provider/dsc_script.rb

```diff
@@ -85,20 +84,24 @@ def run_configuration(operation)
 
         config_manager = Chef::Util::DSC::LocalConfigurationManager.new(@run_context.node, config_directory)
 
-        shellout_flags = {
-          cwd: @dsc_resource.cwd,
-          environment: @dsc_resource.environment,
-          timeout: @dsc_resource.timeout,
-        }
+        cwd = @dsc_resource.cwd || Dir.pwd
+        original_env = ENV.to_hash
 
         begin
-          configuration_document = generate_configuration_document(config_directory, configuration_flags)
-          @operations[operation].call(config_manager, configuration_document, shellout_flags)
+          ENV.update(@dsc_resource.environment || original_env)
```

Comment:
```suggestion
          ENV.replace(@dsc_resource.environment) if @dsc_resource.environment
```

* I'm not sure what side effects this could have, but given that avoiding the method call is just as easy, this feels midly safer.
* `.update` will merge the provided env variables into the existing ones. Is that what is intended? This appears to be undocumented so I'm not sure.

Reply:
yeah we want the merge (`update`). Previously we shelled out to powershell and passed the `environment` as a shell arg. This does the equivalent in proc.